src/app/Layout/Layout.tsx

```diff
@@ -0,0 +1,74 @@
+"use client"
+import React, {  useState } from 'react'
+import {
+    MagnifyingGlassIcon,
+    HamburgerMenuIcon,
+} from '@radix-ui/react-icons'
+import { Input } from "@/components/ui/input"
+import { Button } from "@/components/ui/button"
+import { Sidebar } from '../components/Sidebar'
+import { usePathname } from 'next/navigation'
+import { RoutesTitle } from '@/lib/routes'
+
+type UserRole = "Incharge" | "member" | "Shift-Incharge";
+
+export function Layout({ children }: { children: React.ReactNode, }) {
+    const pathname = usePathname()
+
+    const [sidebarOpen, setSidebarOpen] = useState(false)
+    const [isTilleHide, setIsTitleHide] = useState(false)
+    const toggleSidebar = () => setSidebarOpen(!sidebarOpen)
```

Comment:
We can't mix function handler in between `useStates`. Very hard to read between lines.

Reply:
fixed